pkg/extension/sumologicextension/extension.go

```diff
@@ -57,50 +61,85 @@ const (
 )
 
 func newSumologicExtension(conf *Config, logger *zap.Logger) (*SumologicExtension, error) {
+	collectorName := ""
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+	credentialsGetter := credsGetter{
+		conf:   conf,
+		logger: logger,
+	}
 	if conf.CollectorName == "" {
-		return nil, errors.New("collector name is unset")
+		key := createHashKey(conf)
+		// If collector name is not set by the user check if the collector was restarted
+		if !credentialsGetter.CheckCollectorCredentials(key) {
+			// If credentials file is not stored on filesystem generate collector name
+			collectorName = fmt.Sprintf("%s-%s", hostname, uuid.New().String())
```

Comment:
```suggestion
			collectorName = fmt.Sprintf("%s-%s", hostname, uuid.New())
```

If a struct implements [`Stringer`](https://golang.org/pkg/fmt/#Stringer) then no need to add `.String()` call in format string for `%s`